src/pages/api/progress.ts

```diff
@@ -0,0 +1,85 @@
+import TinCan from 'tincanjs'
+import axios from 'axios'
+import {NextApiRequest, NextApiResponse} from 'next'
+import {connectScrollTo} from 'react-instantsearch-dom'
+
+const SCORM_CLOUD_ENDPOINT = `https://cloud.scorm.com/lrs/PZ9CYEKKV8/`
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  console.log(req.query)
+  if (req.method === 'GET') {
+    try {
+      const lrs = new TinCan.LRS({
+        endpoint: SCORM_CLOUD_ENDPOINT,
+        username: process.env.SCORM_CLOUD_USERNAME,
+        password: process.env.SCORM_CLOUD_PASSWORD,
+        allowFail: false,
+      })
+
+      const verb = req.query.verb
+
+      const statement: any = new TinCan.Statement({
+        actor: {
+          account: {
+            homePage: 'https://egghead.io',
+            name: '1',
+          },
+        },
+        verb: {
+          id: `https://w3id.org/xapi/dod-isd/verbs/${verb}`,
+          display: {
+            'en-US': verb,
+          },
+        },
+        target: {
+          id: 'egghead::lesson::420',
+          definition: {
+            name: {
+              'en-US': 'Taming the Shrew',
+            },
+            description: {
+              'en-US': 'This is a really great video lesson',
+            },
+            type: 'https://egghead.io/xapi/types#video',
```

Comment:
these `IRI` identifiers look like a URL, but they aren't. They explain it in the spec in detail, but effectively this is a way to associate OUR resource types with events. Basically "this actor did this to this thing"